### PR TITLE
Add rolling metric calculator for pearson r

### PR DESCRIPTION
**Pull request by sebhahn**
It is working, though I'm not entirely happy about the implementation. e.g. pandas doesn't make it easy to use `apply` on multiple columns. I'll try to develop another solution using numba

### CODE REVIEW

**Comment by sebhahn** [comment]
@wpreimes could you please test the rolling calculator in your test environment?

**Comment by wpreimes** [comment]
Great, yes I will review and test this

**Comment by sebhahn** [comment]
@wpreimes please have a look at the latest implementation. if you are happy you can merge it. there is now only one method for pearson r because there seems to be no performance difference any longer

File: requirements.txt [hunk 1]
```diff
@@ -10,4 +10,5 @@ ascat>=1.0
 ismn>=0.3
 configparser
 pykdtree
-repurpose>=0.7.1
\ No newline at end of file
+repurpose>=0.7.1
```
Comment (wpreimes) [inline requirements.txt:13]: Do we keep the requirements.txt file with the new structure`?

File: pytesmo/validation_framework/metric_calculators.py [hunk 2]
```diff
@@ -28,15 +28,19 @@
 Metric calculators implement combinations of metrics and structure the output.
 """
 
+import copy
+import itertools
+import pandas as pd
+from pandas.api.indexers import BaseIndexer
+import numpy as np
+from numba import jit
```
[the remark this answers is not in the document]
Comment (wpreimes) [reply pytesmo/validation_framework/metric_calculators.py:37]: Done

File: requirements.txt (hunk 1 above)
Comment (sebhahn) [reply requirements.txt:13]: this will be gone once pyscaffold merge request is done

**Comment by wpreimes** [comment]
Except for the failing test, let me check that...

**Comment by wpreimes** [comment]
@sebhahn Can you add RMSD to the ts_vars in the test when writing the results, then it should pass https://travis-ci.org/github/TUW-GEO/pytesmo/jobs/689222689#L1493

**Pull request merged**